### DISCUSSION TITLE
Option to display page titles

### DISCUSSION THREAD

**Discussion by rvlak** [plausible/analytics#1450] [Anything Else]
The 'Top pages' report can only show URLs, which makes it a bit cumbersome to use for reporting back to the rest of the organisation. For a simpel report on e.g. our Top 25 pages, I'd have to manually look up and copy 25 page titles into an Excel. Would it be possible to add a toggle switch or menu option to show 'Titles' instead of 'URLs'?

**Comment by metmarkosaric** [comment]
thanks @rvlak! haven't heard this request before. could be interesting to experiment by allowing to for instance click on "page url" header to change it to show "page title" instead. or allow the export of url + title in our CSV. is there any specific reason that you're reporting by page title rather than the url?

**Reply by rvlak** [reply]
Hi @metmarkosaric Thanks for your kind reply! I compile quarterly reports on our online outreach and share them with my colleagues, and I know from experience that reports with real page titles are much easier for them to grasp / read than having to interpret sometimes cryptical URLs :-)

**Reply by metmarkosaric** [reply]
makes sense, thanks for sharing the details! we will consider this but i cannot make any promise on if or when it will be done. we are a small team with limited development resources so need to focus on most frequently requested features first. thanks for understanding!

**Reply by rvlak** [reply]
Sure, I fully understand!

**Reply by Lucent** [reply]
I would recommend a page report that is based on page title instead of URL. I serve one URL with a vary header that responds with content in different languages. Differentiating by title would allow me to see which languages are most popular. It can also take the place of dealing with hashes since hash updates also usually update the title.

**Comment by ayoisaiah** [comment]
I'm with you @rvlak 
![Screenshot from 2022-05-09 12-48-56](https://user-images.githubusercontent.com/13372163/167408008-6a1bdb9f-fc31-4cfe-9ff9-a2f860eeaf65.png)

**Comment by yaetuz** [comment]
Please consider adding this. The use-case is the same as above.

**Comment by ukutaht** [comment]
Do some people also need both? Or is it a choice between URL vs title?

Meaning, should we collect and surface BOTH the title and url, or would it be ok to just collect one or the other depending on user choice?

**Reply by rvlak** [reply]
HI @ukutaht Ideally we would like to have both (similar to Google Analytics), e.g. with a toggle switch or menu option to show 'Titles' instead of 'URLs'

**Reply by ayoisaiah** [reply]
Yes I agree that would be perfect

**Reply by yaetuz** [reply]
Both would be nice. I have user-generated content. Urls can be nondescriptive. Titles can be non-unique. To quickly assess what's going on combination of url+title would be perfect for me.

Here is a screenshot (taken from google images) of Realtime Google Analytics. It has two columns - page and title for that page.
![image](https://user-images.githubusercontent.com/108496496/176892341-055ed25d-06d5-4f2b-9a5a-56e700298a21.png)

**Comment by thefedoration** [comment]
I need this feature as well, but in my case it's for goal tracking.
I love being able to filter to a goal conversion and have the URLs that this goal is achieved at show up in "Top Pages".
This would be even more useful with a `page title` view, as many of my goals are achieved at dynamic URLs.

**Comment by sptv2reg** [comment]
I would also like to see this although for a different reason.

Our journalists change the titles a lot resulting in lot's of different nice URLs. These end up as different pages in Plausible. I can use custom location but that doesn't really work because that would make the path unintelligible (like for example a UUID).

So it would be great if I could:

1) Use a custom location, like domain.com/article/[UUID] - this would ensure that all hits would be attributed to the same article (this is already possible)
2) Ability to show Titles in the UI to make it intelligible for the end users

Ps. I assume that you use the URL as the primary key in the database (for the above to work).

**Comment by peteruithoven** [comment]
Note that contrary to for example Google analytics or Matomo the title currently doesn't seem to be included in what's send to Plausible on a pageview. So Plausible currently does not have the data to display titles.

**Reply by metmarkosaric** [reply]
yes, that's correct @peteruithoven! we don't collect this at the moment so would need to change how Plausible works to support it. there are also storage concerns as this would increase our storage needs/costs which is why it's not that simple feature to add

**Comment by assmith** [comment]
While it wouldn't suit everyone, one option that would allow retrospective naming of urls and not require additional storage would be Goal Names for Pageview Goals #838. It would mean that you needed to manually define a goal for each url, so not ideal for very large sites, or sites with highly dynamic urls. I don't think it's a replacement for this feature request, but it might be a good stop gap for many use cases, particularly becasue you can group pages using wildcards. If you could then use OR logic #3122 it would make Goals quite a good swiss army knife for reporting to non-technical team members which was one of the original stated goals of this feature request.

**Comment by chrisarmstrong** [comment]
Also needing this... are there any workarounds currently?

**Reply by chrisarmstrong** [reply]
Guessing there’s been no change on this?

**Reply by ghost** [reply]
On the face of it, they don't seem to have decided to implement this essential function. For your information, Umami does.

**Comment by GitFrontier** [comment]
Is it safe to assume that this has yet to be added or is even on the roadmap? I am struggling to figure out how to add this but need it. We track app + site, with rewritten short urls and knowing the names like we can get in Umami or google is necessary. 

<img width="377" height="280" alt="image" src="https://github.com/user-attachments/assets/2a5c5cb4-ae51-44e2-a4f0-e9fc13cf959d" />
